File: mofuzz-mf-distributed/mofuzz-document-server-frontend/src/app/model/result-statistic.ts
export interface ResultStatistic {
  crashResults: number;
  hangResults: number;
  regularResults: number;
  uniqueExceptions: ExceptionCount[];
  longestDuration: ResultDuration;
  shortestDuration: ResultDuration;
  averageDuration: number;
}

export interface ExceptionCount {
  exception: string;
  count: number;
}

export interface ResultDuration {
  file: string;
  duration: number;
}

export interface ClientResultCount {
  client: string;
  count: number;
}

File: mofuzz-mf-distributed/mofuzz-document-server/src/main/java/de/hub/mse/server/service/analysis/ResultStatistic.java
package de.hub.mse.server.service.analysis;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class ResultStatistic {
    private int crashResults;
    private int hangResults;
    private int regularResults;

    private List<ExceptionCount> uniqueExceptions;

    private ResultDuration longestDuration;
    private ResultDuration shortestDuration;
    private long averageDuration;
}
